homeassistant/components/shelly/climate.py

```diff
@@ -0,0 +1,205 @@
+"""Climate support for Shelly."""
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any, Final, cast
+
+from aioshelly.block_device import Block
+import async_timeout
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_OFF,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.components.shelly import BlockDeviceWrapper
+from homeassistant.components.shelly.entity import ShellyBlockEntity
+from homeassistant.components.shelly.utils import get_device_entry_gen
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.restore_state import RestoreEntity
+
+from .const import (
+    AIOSHELLY_DEVICE_TIMEOUT_SEC,
+    BLOCK,
+    DATA_CONFIG_ENTRY,
+    DOMAIN,
+    SHTRV_01_TEMPERATURE_SETTINGS,
+)
+
+_LOGGER: Final = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up climate device."""
+
+    if get_device_entry_gen(config_entry) == 2:
+        return
+
+    wrapper = hass.data[DOMAIN][DATA_CONFIG_ENTRY][config_entry.entry_id][BLOCK]
+    blocks = [block for block in wrapper.device.blocks if hasattr(block, "targetTemp")]
```

Comment:
Do we expect to have here more than one block? As far as I understand we only search for the block which contains the `targetTemp`
We should probably just loop until we find the block or return